src/sidebar/store/util.js

```diff
@@ -20,42 +20,62 @@ function actionTypes(updateFns) {
 function createReducer(...actionToUpdateFn) {
   // Combine the (action name => update function) maps together into a single
   // (action name => update functions) map.
+  //
+  // After namespace migration, remove the requirement for actionToUpdateFns
+  // to use arrays. Why? createReducer will be called once for each module rather
+  // than once for all modules.
   const actionToUpdateFns = {};
   actionToUpdateFn.forEach(map => {
     Object.keys(map).forEach(k => {
       actionToUpdateFns[k] = (actionToUpdateFns[k] || []).concat(map[k]);
     });
   });
 
-  return (state, action) => {
+  return (state = {}, action) => {
     const fns = actionToUpdateFns[action.type];
     if (!fns) {
       return state;
     }
+    // Some states return an array rather than an object. They need to be
+    // handled differently so we don't cast them to an object.
+    if (Array.isArray(state)) {
+      return [...fns[0](state, action)];
```

Comment:
The array will go away once we move all modules to namespaces. Array types are only used for namespaces modules and they can only have 1 item in the array. Using `[0]` is okay here for now until we remove the array altogether.